test/parallel/test-fs-write-file-uint8array.js

```diff
@@ -17,13 +17,18 @@ const s = '南越国是前203年至前111年存在于岭南地区的一个国家
           '历经五代君主。南越国是岭南地区的第一个有记载的政权国家，采用封建制和郡县制并存的制度，' +
           '它的建立保证了秦末乱世岭南地区社会秩序的稳定，有效的改善了岭南地区落后的政治、##济现状。\n';
 
-const input = Uint8Array.from(Buffer.from(s, 'utf8'));
+const inputBuffer = Buffer.from(s.repeat(8), 'utf8');
 
-fs.writeFileSync(filename, input);
-assert.strictEqual(fs.readFileSync(filename, 'utf8'), s);
+for (const expectView of common.getArrayBufferViews(inputBuffer)) {
+  fs.writeFileSync(filename, expectView);
+  assert.strictEqual(fs.readFileSync(filename, 'utf8'), expectView);
```

Comment:
`fs.readFileSync(..., 'utf8')` returns a string, so the assertion should compare with `s.repeat(8)` instead.

Reply:
ah! indeed. so that will work for reads that are string based, that is, for `utf8` strings,  `Buffer`s, `Uint8Array`s. How should I test for the other typed arrays and data views?

For instance, when encoding is not specified `fs.readFileSync()` returns a Buffer. How do i match this with `Int8Array`s or other typed arrays?

Reply:
You don't. The test is testing if `writeFile` supports non-`Buffer` TypedArrays. It's not testing `readFileSync`, and you should always compare the result of `readFileSync` to the string.